### ISSUE TITLE
LM algos throw ConnectionNotFound exception instead of setting Path#found=false

### ISSUE THREAD

**Issue by easbar** [graphhopper/graphhopper#1865]
Just a minor inconsistency, should this be fixed?

ca6d812c2b379319237e9460fc53336f544baf83

**Comment by karussell** [comment]
The ConnectionNotFound is more precise IMO, but isn't possible to be done fast also from e.g. Dijkstra. I'm unsure if we should reduce the "error message precision" of LM because of that.

Or we introduce some subnetwork ID for all algorithms...

**Comment by easbar** [comment]
I just mean `Path RoutingAlgorithm#calcPath` should be implemented the same way by all algorithms. Either they should all throw `ConnectionNotFound` or all return a `Path` with `Path#found=false`.

**Comment by easbar** [comment]
> Or we introduce some "subnetwork ID lookup" for all algorithms...

Yes I think we agreed on saving the subnetwork ID with every node (to deal with some other problem I do not remember atm)

**Comment by karussell** [comment]
> ConnectionNotFound (probably would have to catch it in some places, but I do not think this is a bad thing)

I do not really like checked exceptions and so I prefer you suggestion about an error object which we could fetch in case of found==false. 

> Yes I think we agreed on saving the subnetwork ID with every node

Ah, yes, this would be good and does not cost much (<0.5 GB for world wide setup) IF we reduce the ID count to e.g. 128 like we do for LM. Which means some smaller subnetworks get a "0" as there are too many, but this would still help in mostly all scenarios.

**Comment by easbar** [comment]
> I do not really like checked exceptions and so I prefer you suggestion about an error object which we could fetch in case of found==false.

I agree. Changing all other algorithms to throw a (checked) exception seems too much effort for the minor issue of this more detailed error message, and so far everyone seems to be happy with having to check `path.isFound()`.
